layer/projects/project_runner.py

```diff
@@ -160,6 +165,33 @@ def run(self, debug: bool = False, printer: Callable[[str], Any] = print) -> Run
                     )
         return run
 
+    async def _upload_tar_packages(self, client: LayerClient) -> None:
+        trace_config = aiohttp.TraceConfig()
+        async with aiohttp.ClientSession(
+            raise_for_status=True, trace_configs=[trace_config]
+        ) as session:
+            upload_tasks = [
+                self._upload_tar_package(client, definition, session)
+                for definition in self.definitions
+            ]
+            await asyncio.gather(*upload_tasks)
+
+    async def _upload_tar_package(
+        self,
+        client: LayerClient,
+        function: FunctionDefinition,
+        session: aiohttp.ClientSession,
+    ) -> None:
+        with open(function.tar_path, "rb") as package_file:
+            presigned_url = client.executor_service_client.get_upload_path(
+                self.project_full_name, f"{function.asset_name}.tar"
+            )
+            await session.put(
+                presigned_url,
+                data=package_file,
+                timeout=aiohttp.ClientTimeout(total=None),
```

Comment:
Is it intentional we don't enforce any timeout there ?

Reply:
Not really, I c/p from another place (and we don't have timeout in either of those places)